pkg/collector/check/check.go

```diff
@@ -27,3 +27,9 @@ type Check interface {
 	GetWarnings() []error                                // return the last warning registered by the check
 	GetMetricStats() (map[string]int64, error)           // get metric stats from the sender
 }
+
+// PythonCheck extends the Check interface with specific method for the python checks
+type PythonCheck interface {
+	Check
+	Version() string // return the version of the check if available
+}
```

Comment:
would it be more future proof to add `Version` to `Check` and just return the agent version for core checks? This way you'd also save a type assertion when creating the `Stats`.

Reply:
Good question. On one side that force us to extend the API for every check in the agent and it would display the agent version for every Go check in the status page (is it useful ?). On the other side we avoid a type assertion and a new interface. 

If feel that having a `PythonCheck` interface could be useful going forward (a python check is more complex than a go one: wheel and stuff) and the type assertion is only done at the check creation, so no impact on performances.

I don't have a strong opinion, so if people think extending the `Check` interface is better I'll do it.

Reply:
I kind of agree with the notion that `Version` being an attribute of all checks, not just python checks is more future-proof. It's also true the `PythonCheck` interface might make sense in the future, but maybe not specifically for this particular case. I don't have a die-for strong opinion, but I agree with @masci argument.